lib/src/main/java/me/onebone/toolbar/CollapsingToolbarScaffold.kt

```diff
@@ -45,6 +49,78 @@ class CollapsingToolbarScaffoldState(
 		get() = offsetYState.value
 
 	internal val offsetYState = mutableStateOf(initialOffsetY)
+
+	// TODO: Maybe should move toolbar expand/collapse methods to CollapsingToolbarState
+	//  but offset expand/collapse leave in CollapsingToolbarScaffoldState?
+
+	// TODO: A strange jump in snap speed is often observed
+	@ExperimentalToolbarApi
+	suspend fun expand(duration: Int = CollapsingToolbarDefaults.ExpandDuration) {
+		val anim = AnimationState(toolbarState.height.toFloat())
+
+		toolbarState.scroll {
+			var prev = anim.value
+			anim.animateTo(toolbarState.maxHeight.toFloat(), tween(duration)) {
+				scrollBy(value - prev)
+				prev = value
+			}
+		}
+	}
+
+	// TODO: A strange jump in snap speed is often observed
+	@ExperimentalToolbarApi
+	suspend fun collapse(duration: Int = CollapsingToolbarDefaults.CollapseDuration) {
+		val anim = AnimationState(toolbarState.height.toFloat())
+
+		toolbarState.scroll {
+			var prev = anim.value
+			anim.animateTo(toolbarState.minHeight.toFloat(), tween(duration)) {
+				scrollBy(value - prev)
+				prev = value
+			}
+		}
+	}
+
+	@ExperimentalToolbarApi
+	suspend fun expandOffset(snapStrategy: SnapStrategy) {
+		val anim = AnimationState(offsetYState.value.toFloat())
+
+		anim.animateTo(0f, tween(snapStrategy.expandDuration)) {
+			offsetYState.value = value.toInt()
+		}
+	}
+
+	@ExperimentalToolbarApi
+	suspend fun collapseOffset(snapStrategy: SnapStrategy) {
+		val anim = AnimationState(offsetYState.value.toFloat())
+
+		anim.animateTo(-toolbarState.minHeight.toFloat(), tween(snapStrategy.collapseDuration)) {
+			offsetYState.value = value.toInt()
+		}
+	}
```

Comment:
I think it would be more flexible to use if its parameter is duration instead. `CollapsingToolbarScaffoldState` seems to have no need to know anything about `SnapStrategy`.